### ISSUE TITLE
::init() could return an Option

### ISSUE THREAD

**Issue by Yamakaky** [Immington-Industries/rust-wlc#14]
The example could be:

``` rust
let interface = ...;
rustwlc::init(interface).expect("Unable to initialize!").run();
```

It would be type-safer. It could be improved to prevent double initialisation.

**Comment by Timidger** [comment]
Ah yes, I had been meaning to do that but I guess it slipped through the cracks. Thanks for the reminder to return an Option.

I like the syntax you put there for run, but that will require a bit more work unfortunately to ensure that run can't be called before init. I've pushed the changes in the most recent commit ([53ae441](https://github.com/Immington-Industries/rust-wlc/commit/53ae441f9d5790513841ed87b2e2699504f46cf7)) to at least return an Option. 

I'll keep this open until we can get run to only work if init has been called first.

**Comment by Timidger** [comment]
Fixed with latest commit  4a3871a . Returns a pointer to the run function

**Comment by Yamakaky** [comment]
Well, why not ^^
